docs/source/task-launch.rst

```diff
@@ -84,67 +117,122 @@ We can submit tasks from other tasks.  This allows us to make decisions while
+    * If using Dask objects, will perform some graph merging to perform less
+      computation and communication. See "`The compute function`_" for more
+      detail.
+* ``get_client``
+    * Will continue to occupy a scheduling slot, unless ``secede`` and
+      ``rejoin`` are called (otherwise may deadlock the cluster!).
+    * The worker running the function will possibly transition to a different
+      scheduling slot (the worker that submitted the jobs is labeled as "maybe
+      long running").
+* ``worker_client``
+    * Intended to submit long running jobs (the worker is labeled as "long
+      running")
+    * Implemented as a context manager, which provides some ease of use.
+    * Will ``secede`` from the current worker, and likely transition to another
+      worker.
+    * Establishing a connection to the scheduler takes roughly 10–20ms. It's
+      wise for the submitted jobs to be several times longer than this.
+
+Transitioning to another worker involves moving all of the functions internal
+state to another machine. The time cost of this will not be insignificant if a
+large array is present in memory.
+
+For more details on transitioning to different workers, see "`Scheduling
+Policies`_" and "`Scheduling State`_".
+
+.. _Scheduling Policies: https://distributed.readthedocs.io/en/latest/scheduling-policies.html
+.. _Scheduling State: https://distributed.readthedocs.io/en/latest/scheduling-state.html
+.. _The compute function: https://dask.pydata.org/en/latest/scheduler-overview.html#the-compute-function
+
+``dask.compute``
+~~~~~~~~~~~~~~~~
+
+``dask.compute`` behaves as normal: it submits the functions to the graph,
+optimizes for less bandwidth/computation and gathers the results.
 
 .. code-block:: python
 
-   from distributed import worker_client
+    from distributed import Client
+    from dask import delayed, compute
 
-   def process_all(data):
-       with worker_client() as e:
-           elements = e.scatter(data)
-           futures = e.map(process_element, elements)
-           analysis = e.submit(aggregate, futures)
-           result = analysis.result()
-       return result
+    @delayed
+    def fib(n):
+        if n < 2:
+            return n
+        a, b = compute(fib(n-1), fib(n-2))
+        return a + b
 
-    analysis = e.submit(process_all, data)  # spawns many tasks
+    client = Client()  # to change default dask scheduler
+    assert fib(4).compute() == 3
 
-This approach is somewhat complex but very powerful.  It allows you to spawn
-tasks that themselves act as potentially long-running clients, managing their
-own independent workloads.
+Look at the documentation of ``dask.compute`` for more detail.
 
-Extended Example
-~~~~~~~~~~~~~~~~
+``get_client``
+~~~~~~~~~~~~~~
 
-This example computing the Fibonacci numbers creates tasks that submit tasks
-that submit tasks that submit other tasks, etc..
+``get_client`` is a lower-leve interface than ``worker_client``. It exposes
```

Comment:
typo `leve` -> `level`.  

We haven't yet gotten to `worker_client` and so shouldn't mention it.  We also haven't gotten to `secede` and `rejoin`.  I recommend doing a simple example with `get_client` alone, saying that it gives us the ability to do all of the client operations like submit, scatter, gather, etc.. from within a task.

Then I would provide a cautionary note about flooding all of the available scheduling slots, and then expand the example with secede and rejoin around the gather to show how they can resolve this problem.